cmd/backport-verifier/server.go

```diff
@@ -0,0 +1,182 @@
+)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The backport validation plugin is used to validate that backports come from merged PRs in a configured upstream repository.`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/validate-backports",
+		Description: "Validate that backports come from merged PRs in the upstream repository",
+		WhoCanUse:   "Anyone",
+		Examples:    []string{"/validate-backports"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	config func() *Config
+
+	ghc githubClient
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) {
+	if !publicizeRe.MatchString(ic.Comment.Body) {
+		return
+	}
+	l.Info("Backport validation of PR has been requested.")
+	s.handle(l, ic.Repo.Owner.Login, ic.Repo.Name, ic.Comment.User.Login, ic.Issue.Number)
+}
+
+func (s *server) handlePullRequestEvent(l *logrus.Entry, event github.PullRequestEvent) {
+	if event.Action != github.PullRequestActionOpened && event.Action != github.PullRequestActionSynchronize {
+		return
+	}
+	l.Info("Changes to pull request require backport validation")
+	s.handle(l, event.Repo.Owner.Login, event.Repo.Name, event.PullRequest.User.Login, event.PullRequest.Number)
+}
+
+func (s *server) handle(l *logrus.Entry, org, repo, user string, num int) {
+	logger := l.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	upstream, configured := s.config().Repositories[fmt.Sprintf("%s/%s", org, repo)]
+	if !configured {
+		if err := s.ghc.CreateComment(org, repo, num, fmt.Sprintf("@%s: no upstream repository is configured for validating backports for this repository.", user)); err != nil {
+			logger.WithError(err).Warn("couldn't create comment")
+		}
+		ensureLabels(s.ghc, l, invalidBackportsLabel, org, repo, num)
+		return
+	}
+
+	parts := strings.Split(upstream, "/")
+	upstreamOrg, upstreamRepo := parts[0], parts[1]
+
+	commits, err := s.ghc.ListPRCommits(org, repo, num)
+	if err != nil {
+		if commentErr := s.ghc.CreateComment(org, repo, num, fmt.Sprintf(`@%s: could not list commits in this pull request. Please try again with /validate-backports.
+
+<details>
+
+%s
+
+</details>`, user, err)); commentErr != nil {
+			logger.WithError(commentErr).Warn("couldn't list commits")
+		}
+	}
+
+	invalidCommits := map[string]string{}
+	validCommits := map[string]string{}
+	upstreamPullsByCommit := map[string]int{}
+	errorsByCommit := map[string]string{}
+	for _, commit := range commits {
+		if parts := upstreamPullRe.FindStringSubmatch(commit.Commit.Message); len(parts) > 0 {
+			pr, err := strconv.Atoi(parts[1])
+			if err != nil {
+				// based on the regex this should not happen ...
+				logger.WithError(err).Warn("Failed to parse PR as integer")
+				errorsByCommit[commit.Commit.SHA] = fmt.Sprintf("failed to parse PR identifier: %s", err.Error())
```

Comment:
IMHO this is `invalidCommits`, not `errorsByCommit`

Reply:
It can't happen and would be a bug in my code, an error in handling the content of the repo IMO. In any case I think it's impossible given the regex ... unless I guess they put in something larger than int64? lol